mirgecom/initializers.py

```diff
@@ -610,6 +792,7 @@ def exact_rhs(self, discr, q, t=0.0):
         mass[:] = 1.0
         massrhs = 0.0 * mass
         energyrhs = 0.0 * mass
-        momrhs = make_obj_array([0 * mass])
+        momrhs = make_obj_array([0 * mass for i in range(self._dim)])
+        yrhs = make_obj_array([0 * mass for i in range(self._nspecies)])
 
-        return flat_obj_array(massrhs, energyrhs, momrhs)
+        return flat_obj_array(massrhs, energyrhs, momrhs, yrhs)
```

Comment:
(Question: When should `join_conserved` be used vs. `flat_obj_array`?)

Reply:
I can't actually think of a good reason to use `flat_obj_array` inside the Euler module outside of tests. This particular instance (I believe) is fixed up in the PyrometheusEOS PR sitting just above this one.

Reply:
[Just wanted to remind myself that these were indeed changed.](https://github.com/illinois-ceesd/mirgecom/pull/147/files#diff-98e0920213ef8557c3ac6ded9d1dbf50a9a6c7f6de1b7b998fe0891c3de54ee1)